master/buildbot/status/buildstep.py

```diff
@@ -375,12 +387,21 @@ def asDict(self):
         result['times'] = self.getTimes()
         result['expectations'] = self.getExpectations()
         result['eta'] = self.getETA()
-        result['urls'] = self.getURLs()
         result['step_number'] = self.step_number
         result['hidden'] = self.hidden
+
+        args = getCodebasesArg(request)
         result['logs'] = [[l.getName(),
-            self.build.builder.status.getURLForThing(l)]
+            self.build.builder.status.getURLForThing(l) + args]
                 for l in self.getLogs()]
+
+        result["urls"] = self.getURLs()
+
+
+        if request is not None:
+            from buildbot.status.web.base import path_to_step
+            result['url'] = path_to_step(request, self)
```

Comment:
Hi Stibbons, I actually created the PR by mistake, but do feel free to look it over